File: Go/Kex_2.go
package main

import (
	"fmt"
	"math/rand"
	"strconv"
)

type Kex_2 struct {

}

func (kex2 Kex_2) Execute() {
	fmt.Println("冒険が今始まる！")
	player := Player{
		Hp: PlayerHp,
		Defense: PlayerDefense,
	}
	continueGame := 1
	kills := 0

	for continueGame != 0 {
		fmt.Printf("現HP：%d\n", player.Hp)
		fmt.Print("奥に進みますか？（１：奥に進む　０．帰る）＞")
		var reader string
		_, _ = fmt.Scanln(&reader)
		continueGame, _ = strconv.Atoi(reader)
		if continueGame == 1 {
			typeNo := rand.Intn(3)
			enemy := CreateEnemy(typeNo)
			result := AttackEnemy(&player, &enemy)
			kills++
			if !result {
				break
			}
		}
	}

	fmt.Printf("リ〇ミト！\n\n")
	fmt.Printf("戦闘回数：%d回　残りHP：%d\n", kills, player.Hp)
}

File: CSharp/Kex_2.cs
using System;

namespace CSharp
{
    public static class Kex_2
    {
        public static void Execute()
        {
            Random random = new Random();
            Console.WriteLine("冒険が今始まる！");
            PlayerEx player = new PlayerEx(PlayerEx.INITIAL_HP);
            bool continueGame = true;
            int kills = 0;

            do
            {
                Console.WriteLine($"\n現HP：{player.Hp}");
                Console.Write("奥に進みますか？（１：奥に進む　０．帰る）＞");
                continueGame = Convert.ToBoolean(int.Parse(Console.ReadLine()));

                if (continueGame)
                {
                    int type = random.Next(0, 3);
                    Enemy enemy = Kex_2_Enemy.CreateEnemy((EnemyType)type);
                    bool result = Kex_2_Battle.AttackEnemy(ref player, ref enemy);
                    kills++;

                    if (!result)
                    {
                        break;
                    }
                }

            } while (continueGame);

            Console.WriteLine("リ〇ミト！\n");
            Console.WriteLine($"戦闘回数：{kills}回　残りHP：{player.Hp}");
        }
    }
}
